### ISSUE TITLE
Fix for cached images (new upload system)

### ISSUE THREAD

**Issue by MetinSeven** [LinkStackOrg/LinkStack#480]
When I change my Linkstack page's custom background, instead of properly refreshing, one of my previously uploaded background images reappears, in the background image preview, the theme preview and the link page itself.

**Comment by JulianPrieber** [comment]
An easy fix is to disable caching in your Advanced Config, add this to the analytics section:
```
<meta http-equiv="Cache-Control" content="no-cache, no-store, must-revalidate">
<meta http-equiv="Pragma" content="no-cache">
<meta http-equiv="Expires" content="0">
```

**Comment by JulianPrieber** [comment]
We'll implement a long time fix by generating unique file names to still have caching, but new uploads will show instantly.

**Comment by MetinSeven** [comment]
> We'll implement a long time fix by generating unique file names to still have caching, but new uploads will show instantly.

That's fine as a workaround. Hope to see that change soon, thanks!

**Comment by MetinSeven** [comment]
> An easy fix is to disable caching in your Advanced Config, add this to the analytics section:
> 
> ```
> <meta http-equiv="Cache-Control" content="no-cache, no-store, must-revalidate">
> <meta http-equiv="Pragma" content="no-cache">
> <meta http-equiv="Expires" content="0">
> ```

Thanks! Where can I find the Advanced Config?

**Comment by JulianPrieber** [comment]
Right at the top like this:

```
$analytics = 

/*
|--------------------------------------------------------------------------
| Analytics
|--------------------------------------------------------------------------
|
| Add external analytics services to your LinkStack instance by adding them below.
| Everything you enter below will be added to the <head> tag of every page.
| Formatting in plain HTML is expected.
|
*/

<<<EOD
<!----------Insert your analytics code here:---------->

<meta http-equiv="Cache-Control" content="no-cache, no-store, must-revalidate">
<meta http-equiv="Pragma" content="no-cache">
<meta http-equiv="Expires" content="0">

<!--------------------------------------------------->
EOD;;
```

**Comment by MetinSeven** [comment]
> Right at the top like this:
> 
> ```
> $analytics = 
> 
> /*
> |--------------------------------------------------------------------------
> | Analytics
> |--------------------------------------------------------------------------
> |
> | Add external analytics services to your LinkStack instance by adding them below.
> | Everything you enter below will be added to the <head> tag of every page.
> | Formatting in plain HTML is expected.
> |
> */
> 
> <<<EOD
> 
> 
> <meta http-equiv="Cache-Control" content="no-cache, no-store, must-revalidate">
> <meta http-equiv="Pragma" content="no-cache">
> <meta http-equiv="Expires" content="0">
> 
> 
> EOD;;
> ```

Thanks. Where can I edit my page's code, is this possible from the CMS?

**Comment by MetinSeven** [comment]
Sorry, but I don't see a link to a Config page anywhere in any Dashboard section. 😊

**Comment by JulianPrieber** [comment]
Are you an administrator?

**Comment by MetinSeven** [comment]
I guess not. It's my account, but I don't know if I've got admin level.

**Comment by MarkBTomlinson** [comment]
It seems this is an admin level fix, so I will wait for the administrator to implement the fix. Thanks!

**Comment by MetinSeven** [comment]
@JulianPrieber, if you are the linksta.cc admin, I'd much appreciate it if you could add the code. Thanks in advance.

**Comment by JulianPrieber** [comment]
> @JulianPrieber, if you are the linksta.cc admin, I'd much appreciate it if you could add the code. Thanks in advance.

That's done

**Comment by MetinSeven** [comment]
> > @JulianPrieber, if you are the linksta.cc admin, I'd much appreciate it if you could add the code. Thanks in advance.
> 
> That's done

Thanks! It worked for a little while, but I wanted to change the background once more, and I'm experiencing the same issues as before, with just about every previous background version appearing, but not the latest uploaded image. Could it be that the cache is reenabled?

**Comment by MarkBTomlinson** [comment]
Same, something is odd, good luck figuring it out.

**Comment by JulianPrieber** [comment]
We might have to turn off our server side cache as well. At the very least in 24 hours all changes should show up.

**Comment by MetinSeven** [comment]
> We might have to turn off our server side cache as well. At the very least, in 24 hours, all changes should show up.

That's great, thanks for your support, appreciated!

**Comment by JulianPrieber** [comment]
Implemented in `v4.1.3-beta-1`

**Comment by JulianPrieber** [comment]
Implemented on [linksta.cc](https://linksta.cc)
